### DISCUSSION TITLE
Access to nested parent relation is inconsistent

### DISCUSSION THREAD

**Discussion by Chedi** [openfga/community#297] [General]
Hi,

While trying to model some nested relation, I noticed that accessing any parent level except the one just above the current level gives me an error. For example

```
model
  schema 1.1

type user

type system
  relations
    define admin: [user]
    define staff: [user] or admin

type organization
  relations
    define system: [system]
  
type resource
  relations
    define system      : system from organization   
    define organization: [organization]
    
    define can_delete: staff from system
```

The error being 
```
`can_delete` is an impossible relation for `resource` (no entrypoint)
```

to fix it, I have to redefine the relation in the parent model

```
model
  schema 1.1

type user

type system
  relations
    define admin: [user]
    define staff: [user] or admin

type organization
  relations
    define system: [system]
    define staff : staff from system
    
type resource
  relations
    define organization: [organization]   
    define can_delete: staff from organization
```

To me both are expressing the same intention and I find it cumbersome to have to redefine the relation throughout all the types. I'm wrong to assume that ?

**Comment by rhamzeh** [comment]
Going to convert this to a bug, as the error message is incorrect, it should be:
```
`system` relation used inside from allows only direct relation.
```

But the validation is working, as in the top model is not allowed, you have to redefine the relation in the parent model.

The referenced base relation used after the from `system` in `staff from system` MUST be a directly assignable relation that has no rewrites and does not allow direct assignment of `<type>:*` and `<type>#<relation>`, so it MUST be of the form:
```
define system: [<type>, <type2>, etc..]
```

If the DSL allowed the first, it would not work as you imagine, as the API will not know how to handle it, and implementing that work has been previously judged as a large complicated body of work that we don't plan on doing (at the moment).

**Reply by rhamzeh** [reply]
Filed the bug @ https://github.com/openfga/language/issues/244
